spark/spark-version/2.0/src/main/scala/org/apache/spark/ml/MlTransformer.scala

```diff
@@ -23,7 +23,7 @@ import org.apache.spark.sql.{DataFrame, Dataset}
  * Extends MlTransformer and override process to gain compatibility with
  * both spark 1.5 and spark 2.0.
  */
-abstract class MlTransformer extends Transformer{
+abstract class MlTransformer extends Model[MlTransformer]{
```

Comment:
@hhbyyh , for your questions,
1) Yes, same as what we've done to transformer, for Estimator we do the common logic in one class and interface difference will be handled in different version folders
2) Agreed, I've done refinement to a) use params b) get rid of using 2-d arrays, it looks more extensible and flexible now.
please help take a look and comment, thanks.